### PR TITLE
Utils for Json Serialization

### PR DESCRIPTION
**Pull request by jesusmartinez-stratio**
Improvement from #167 

This feature allows to put json format data into etcd, as well as receive it in that same format

### CODE REVIEW

**Comment by dgutierrez-stratio** [comment]
Hi! We have refactored this PR too, let us know if you need any help whatsoever!

**Comment by lburgazzoli** [comment]
Can you squash it ?

**Comment by dgutierrez-stratio** [comment]
Sure! Thanks for your time!

File: src/main/java/mousio/etcd4j/EtcdClient.java [hunk 1]
```diff
@@ -19,14 +19,7 @@
...
-import mousio.etcd4j.responses.EtcdAuthenticationException;
-import mousio.etcd4j.responses.EtcdException;
-import mousio.etcd4j.responses.EtcdHealthResponse;
-import mousio.etcd4j.responses.EtcdMembersResponse;
-import mousio.etcd4j.responses.EtcdSelfStatsResponse;
-import mousio.etcd4j.responses.EtcdStoreStatsResponse;
-import mousio.etcd4j.responses.EtcdVersionResponse;
-import mousio.etcd4j.responses.EtcdLeaderStatsResponse;
+import mousio.etcd4j.responses.*;
 import mousio.etcd4j.transport.EtcdClientImpl;
```
Comment (lburgazzoli) [inline src/main/java/mousio/etcd4j/EtcdClient.java]: This change is unrelated to the PR, can we change it back ?

File: src/main/java/mousio/etcd4j/EtcdClient.java [hunk 2]
```diff
@@ -214,13 +207,13 @@ public EtcdKeyPutRequest put(String key, String value) {
...
-  * @param ttl to update key with
-  * @return EtcdKeysRequest
-  */
+   * Refresh a key with new ttl
+   * (without notifying watchers when using etcd 2.3+)
+   *
+   * @param key to refresh
+   * @param ttl to update key with
+   * @return EtcdKeysRequest
+   */
```
Comment (lburgazzoli) [inline src/main/java/mousio/etcd4j/EtcdClient.java]: This change is unrelated to the PR, can we change it back ?

File: src/main/java/mousio/etcd4j/EtcdUtil.java [hunk 3]
```diff
@@ -15,12 +15,26 @@
...
+import mousio.etcd4j.responses.EtcdException;
+import mousio.etcd4j.responses.EtcdKeysResponse;
+import mousio.etcd4j.responses.EtcdKeysResponse.EtcdNode;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.xml.bind.DatatypeConverter;
-import java.util.Date;
+import java.io.IOException;
+import java.util.*;
```
Comment (lburgazzoli) [inline src/main/java/mousio/etcd4j/EtcdUtil.java]: If possible I'd like to keep full qualified imports-

File: src/test/java/mousio/etcd4j/TestFunctionality.java [hunk 4]
```diff
@@ -29,10 +16,18 @@
...
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
```
Comment (lburgazzoli) [inline src/test/java/mousio/etcd4j/TestFunctionality.java]: Please avoid changing the import order

File: src/test/java/mousio/etcd4j/TestFunctionality.java [hunk 5]
```diff
@@ -1,26 +1,13 @@
...
-import mousio.etcd4j.responses.EtcdException;
-import mousio.etcd4j.responses.EtcdHealthResponse;
-import mousio.etcd4j.responses.EtcdKeyAction;
-import mousio.etcd4j.responses.EtcdKeysResponse;
-import mousio.etcd4j.responses.EtcdLeaderStatsResponse;
-import mousio.etcd4j.responses.EtcdMembersResponse;
-import mousio.etcd4j.responses.EtcdSelfStatsResponse;
-import mousio.etcd4j.responses.EtcdStoreStatsResponse;
-import mousio.etcd4j.responses.EtcdVersionResponse;
+import mousio.etcd4j.responses.*;
```
Comment (lburgazzoli) [inline src/test/java/mousio/etcd4j/TestFunctionality.java]: Please keep the full qualified import.

**Comment by dgutierrez-stratio** [comment]
All set. BTW, I'm including new features in JSON deserialization (restoring primitive types and arrays for instance). But I'm not done yet, do you mind if we push these improvements in a further PR?

**Comment by lburgazzoli** [comment]
that's fine with me, can you just have a look at travis build that fails ?

**Comment by dgutierrez-stratio** [comment]
It seems that certain operations with EtcdClient freeze when no connection can be stablished to etcd. Since PRs do not launch dockers, that might be the problem. I'm going to set a timeout for these tests so any further PR won't fail.

**Comment by dgutierrez-stratio** [comment]
Would it be possible to configure Travis for dockerTest task on PRs?

**Comment by lburgazzoli** [comment]
Ah yeah, can you tweak this condition to run docker also for pr ?
 https://github.com/jurmous/etcd4j/blob/master/build.gradle#L26

**Comment by dgutierrez-stratio** [comment]
Sure, I'm not familiar with Travis, shall I remove isPR from the condition? That would imply snapsnot releases. Is it ok if I do that?

**Comment by lburgazzoli** [comment]
better if we do not upload snapshots for prs

**Comment by dgutierrez-stratio** [comment]
Done! should I squash both commits?

**Comment by lburgazzoli** [comment]
Merged :)

**Comment by dgutierrez-stratio** [comment]
Pleasure collaborating with you!
